api/src/main/java/org/openmrs/Person.java

```diff
@@ -45,7 +46,7 @@
  * @see org.openmrs.Patient
  */
 @Audited
-public class Person extends BaseChangeableOpenmrsData {
+public class Person extends BaseCustomizableData<PersonAttribute> implements Customizable<PersonAttribute>{
```

Comment:
What are you trying to do here. Why are you extending  `BaseCustomizableData<PersonAttribute> implements Customizable<PersonAttribute>` ?

Reply:
Before i had an error but i have solved it by allowing PersonAttributeType to extend BaseAttributeType.

what i want to achieve is to enable encapsulation, code reusability, inheritance, polymorphism, and abstraction, which together foster a modular, maintainable, and scalable codebase. These principles not only streamline the development process but also ensure that applications are robust, flexible, and easier to manage and extend.